.github/workflows/build.yml

```diff
@@ -4,9 +4,6 @@ on:
   push:
     branches:
       - master
-  pull_request:
-    branches:
-      - master
 
 jobs:
   build-packages:
```

Comment:
To avoid the authentication issue, I believe we could simply use the `GH_PAT` env variable instead of the `GITHUB_TOKEN`. The PAT (personal access token) needs to be stored once in secrets, but then it will use that consistently instead of trying to impersonate the one pushing (at least, that's my understanding).